src/components/category-selection/category-badge/CategoryBadge.tsx

```diff
@@ -1,31 +1,46 @@
 import classes from './CategoryBadge.module.scss';
 
 import { MouseEvent, TouchEvent, useState } from 'react';
-import CategoryModal from '../shared/CategoryModal';
-import { Category } from '../../data/interfaces';
+import CategoryModal from '../../shared/CategoryModal';
+import { Category } from '../../../data/interfaces';
 import classNames from 'classnames/bind';
 
 const cx = classNames.bind(classes);
 
-const CategoryBadge = ({ id, name, color }: Category) => {
+export interface CategoryBadgeProps extends Category {
+    onClick?: () => void;
+    deletable?: boolean;
+}
+
+const CategoryBadge = ({ 
+    id, 
+    name, 
+    color, 
+    onClick, 
+    deletable 
+}: CategoryBadgeProps) => {
     const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
 
     const handleOnClick = (event: MouseEvent | TouchEvent) => {
         event.stopPropagation();
-        setIsModalOpen(true);
+        // TODO remove monkey patch
+        onClick ? onClick() : setIsModalOpen(true);
     };
 
     return (
         <>
-            <span
+            <div
                 onClick={handleOnClick}
                 className={cx({
                     categoryBadge: true,
                     [`categoryColor${color}`]: color,
                 })}
             >
-                {name}
-            </span>
+                <span>
+                    {name}
+                </span>
+                {deletable && <span>×</span>}  
```

Comment:
Utilize the XCircle component with the weight property set to `fill` to align with the Figma design.
![image](https://github.com/Polarts/feel-tracker/assets/105120312/ca1ad3cb-f5c2-45a1-865f-fd47e0169a99)